app/src/main/java/pm/gnosis/heimdall/ui/safe/recover/safe/submit/RecoveringSafeSubmitFragment.kt

```diff
@@ -45,65 +53,86 @@ class RecoveringSafeSubmitFragment : BaseFragment() {
 
-        // Automatically load on start
-        disposables += layout_recovering_safe_submit_retry.clicks().startWith(Unit)
-            .doOnNext {
-                layout_recovering_safe_submit_retry.isEnabled = false
-                layout_recovering_safe_submit_button.isEnabled = false
+        swipeToRefresh.setOnRefreshListener {
+            loadExecuteInfo()
+        }
 
-            }
-            .switchMapSingle { viewModel.loadRecoveryExecuteInfo(safeAddress).mapToResult() }
-            .observeOn(AndroidSchedulers.mainThread())
-            .subscribeForResult(onNext = {
-                layout_recovering_safe_submit_data_balance_value.text = it.paymentToken.displayString(it.balance)
-                layout_recovering_safe_submit_data_fees_value.text = it.paymentToken.displayString(it.paymentAmount)
-                layout_recovering_safe_submit_data_fees_error.visible(!it.canSubmit)
-                layout_recovering_safe_submit_button.visible(it.canSubmit)
-                layout_recovering_safe_submit_button.isEnabled = it.canSubmit
-                layout_recovering_safe_submit_retry.visible(!it.canSubmit)
-                layout_recovering_safe_submit_retry.isEnabled = !it.canSubmit
-            }, onError = {
-                errorSnackbar(layout_recovering_safe_submit_button, it)
-                layout_recovering_safe_submit_button.visible(false)
-                layout_recovering_safe_submit_retry.isEnabled = true
-                layout_recovering_safe_submit_retry.visible(true)
-            })
+        feesInfo.setOnClickListener {
+            InfoTipDialogBuilder.build(context!!, R.layout.dialog_network_fee, R.string.ok).show()
+        }
 
-        disposables += layout_recovering_safe_submit_button.clicks()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        eventTracker.submit(Event.ScreenView(ScreenId.RECOVER_SAFE_REVIEW))
+
+        disposables += recover_safe_bottom_panel.forwardClicks
             .doOnNext {
-                layout_recovering_safe_submit_retry.isEnabled = false
-                layout_recovering_safe_submit_button.isEnabled = false
+                onLoading(true)
             }
-            .switchMapSingle { viewModel.submitRecovery(safeAddress).mapToResult() }
+            .switchMapSingle { viewModel.submitRecovery(safe).mapToResult() }
             .observeOn(AndroidSchedulers.mainThread())
             .subscribeForResult(onNext = {
                 startActivity(SafeMainActivity.createIntent(context!!, it))
             }, onError = {
-                errorSnackbar(layout_recovering_safe_submit_button, it)
-                layout_recovering_safe_submit_button.isEnabled = true
-                layout_recovering_safe_submit_retry.isEnabled = true
+                errorSnackbar(recover_safe_bottom_panel, it)
+                onLoading(false)
             })
     }
 
+    override fun onResume() {
+        super.onResume()
+        loadExecuteInfo()
```

Comment:
Not sure if it is good to have this in `onResume` (might not be a big issue, but we only clear the disposable in `onStop`, not in `onPause`)

Reply:
nvm, thought this is a long running observable .... but this is only a single, so it doesn't matter